packages/drivers/file-driver/src/fileDocumentStorageService.ts

```diff
@@ -93,24 +93,16 @@ export class FluidFetchReader extends ReadDocumentStorageServiceBase implements
      * Finds if a file exists and returns the contents of the blob file.
      * @param sha - Name of the file to be read for blobs.
      */
-    public async read(sha: string): Promise<string> {
+    public async readBlob(sha: string): Promise<ArrayBufferLike> {
         if (this.versionName !== undefined) {
             const fileName = `${this.path}/${this.versionName}/${sha}`;
             if (fs.existsSync(fileName)) {
-                const data = fs.readFileSync(fileName).toString();
+                const data = fs.readFileSync(fileName);
```

Comment:
Please note that function before returned base64-encoded string, now it returns decoded buffer.
There is a mismatch in expectations here, as it's clear from prior code that file contains base64 encoded content.
Normally, to do safe refactoring, you would need to decode data here to match new expectations.
But you are sort of in luck here, as this code is used to read data out of fetch-tool, and this tool conveniently puts decoded data next to encoded data. All you need to do is replace this:
```
            const fileName = `${this.path}/${this.versionName}/${sha}`;
```
to this:
```
            const fileName = `${this.path}/${this.versionName}/decoded/${sha}`;
```

Reply:
Please self-review other places like that for mismatch in expectations - you know what old data looks like (based on signature of functions) and what new functions return, this should be sufficient to spot any mismatches due to changes.

The harder mismatches to spot are places where we passed base-64-encoded strings through multiple layers, as "string" type could mean base64 encoded or decoded string. For this reason, I believe one of the success criteria for your PR should be almost no calls to decode base64 content - that will ensure that we always pass around decoded strings and do not miss decoding, or do not do double-deciding. The only places where decoding should happen is for legacy workflows (supporting read() APIs) in driver code, and maybe some helpers (thought later might smell trouble)

Reply:
sorry im not following. what is the `decoded` here?
``` const fileName = `${this.path}/${this.versionName}/decoded/${sha}`;```

Reply:
wait , did you mean literally "decoded" ? not a variable?

Reply:
Yes, literally that change how I spelled it.